website/project/views/contributor.py

```diff
@@ -534,71 +527,71 @@ def send_claim_email(email, unclaimed_user, node, notify=True, throttle=24 * 360
     return to_addr
 
 
+def check_email_throttle(node, contributor, throttle=None):
+    throttle = throttle or settings.CONTRIBUTOR_ADDED_EMAIL_THROTTLE
+    contributor_record = contributor.contributor_added_email_records.get(node._id, {})
+    if contributor_record:
+        timestamp = contributor_record.get('last_sent', None)
+        if timestamp:
+            if not throttle_period_expired(timestamp, throttle):
+                return True
+    else:
+        contributor.contributor_added_email_records[node._id] = {}
+
+
 @contributor_added.connect
-def notify_added_contributor(node, contributor, auth=None, throttle=None, email_template='default', *args, **kwargs):
```

Comment:
This function was full of very old weirdness so I simplified it a lot and fixed obviously wrong lines like :  `email_template = getattr(mails, 'CONTRIBUTOR_ADDED_PREPRINT_NODE_FROM_OSF'.format(email_template.upper()))` which isn't really doing what it intends to do.